packages/block-editor/src/store/selectors.js

```diff
@@ -1569,6 +1569,30 @@ export function canMoveBlocks( state, clientIds, rootClientId = null ) {
 	);
 }
 
+/**
+ * Determines if the given block is allowed to be edited.
+ *
+ * @param {Object} state    Editor state.
+ * @param {string} clientId The block client Id.
+ *
+ * @return {boolean} Whether the given block is allowed to be edited.
+ */
+export function canEditBlock( state, clientId ) {
+	const attributes = getBlockAttributes( state, clientId );
+	if ( attributes === null ) {
+		return true;
+	}
+
+	const { lock } = attributes;
+	// No `lock` attribute means we can edit the block.
+	if ( lock === undefined || lock?.edit === undefined ) {
```

Comment:
`lock === undefined` is not necessary here, `lock?.edit === undefined` already covers it

Reply:
Even further – this entire `if` could be removed. `return ! lock?.edit;` has the same effect

Reply:
Done - [4656245](https://github.com/WordPress/gutenberg/pull/39950/commits/46562453c5927fd593f052b075cdd21bc43ab63c)